coreneuron/network/netpar.cpp

```diff
@@ -32,7 +32,9 @@
 #include "coreneuron/mpi/core/nrnmpi.hpp"
 int localgid_size_;
 int ag_send_nspike;
+namespace coreneuron {
 int* nrnmpi_nin_;
+}
```

Comment:
This was because of an ODR clash with NEURON, I think.

Reply:
@olupton : could you bit explain the context for this?

Reply:
`nrnmpi_nin_` is defined at global scope in NEURON: https://github.com/neuronsimulator/nrn/blob/3f2f83d5339223c99f3d710424ffec383d038f27/src/nrnmpi/nrnmpi_def_cinc#L10.

Normally for CoreNEURON we duplicate these variables inside the `coreneuron` namespace, but here we are defining these things at global scope too. This violates the one definition rule, so it's undefined behaviour, and some tool (I think ASan, though I don't have the error message to hand) flagged this.

Looking at it now, some of the other variables here are probably falling into the same trap.